### PR TITLE
[DOCS] Expands GET DFA stats API docs with new phases

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR lists the new phase types in the description of the `phase` property of the GET DFA stats API.

## Preview

[GET DFA stats](http://elasticsearch_56407.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-dfanalytics-stats.html)

### CODE REVIEW

File: docs/reference/ml/df-analytics/apis/get-dfanalytics-stats.asciidoc [hunk 1]
```diff
@@ -477,7 +477,16 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=node-transport-address]
 =====
 `phase`::::
 (string) Defines the phase of the {dfanalytics-job}. Possible phases: 
-`reindexing`, `loading_data`, `analyzing`, and `writing_results`.
+* `reindexing`, 
+* `loading_data`, 
+* `computing_outliers` (for {oldetection} only),
+* `feature_selection` (for {regression} and {classification} only),
+* `coarse_parameter_search` (for {regression} and {classification} only),
+* `fine_tuning_parameters` (for {regression} and {classification} only),
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/get-dfanalytics-stats.asciidoc:485]: we're missing `final_training` here

Comment (szabosteve) [reply docs/reference/ml/df-analytics/apis/get-dfanalytics-stats.asciidoc:485]: Thanks @dimitris-athanasiou, I added to the list.

File: docs/reference/ml/df-analytics/apis/get-dfanalytics-stats.asciidoc [hunk 2]
```diff
@@ -477,7 +477,17 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=node-transport-address]
...
+* `final_training` (for {regression} and {classification} only),
+* `writing_results`.
+
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/get-dfanalytics-stats.asciidoc:488]: I think you need a continuation character here to get the following paragraph nested properly:
```suggestion
+
```

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: ebe1e4c
7.8: c7b2652

**Pull request merged**